llvm/utils/TableGen/DAGISelMatcherEmitter.cpp

```diff
@@ -381,8 +381,8 @@ static void EndEmitFunction(raw_ostream &OS) {
 
 void MatcherTableEmitter::EmitPatternMatchTable(raw_ostream &OS) {
 
-  assert(isUInt<16>(VecPatterns.size()) &&
-         "Using only 16 bits to encode offset into Pattern Table");
+  assert(isUInt<32>(VecPatterns.size()) &&
+         "Using only 32 bits to encode offset into Pattern Table");
```

Comment:
This probably should have been / should be report_fatal_error if it depends on the number of patterns in the target

Reply:
Will do a fixup with this change - because I agree, we need something to error regardless of whether we're optimising or not.